storage/badger/transaction/deferred_block_persist.go

```diff
@@ -0,0 +1,271 @@
+package transaction
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// TODO: maybe move this into the package `storage/badger/transaction` (?)
+
+// DeferredBlockPersistOp is a shorthand notation for an anonymous function that takes the ID of
+// a fully constructed block and a `transaction.Tx` as inputs and runs some database operations
+// as part of that transaction. It is a "Future Pattern", essentially saying:
+// once we have the completed the block's construction, we persist data structures that are
+// referenced by the block or populate database indices. This pattern is necessary, because
+// internally to the protocol_state package we don't have access to the candidate block ID yet because
+// we are still determining the protocol state ID for that block.
+type DeferredBlockPersistOp func(blockID flow.Identifier, tx *Tx) error
+
+var noOpPersist DeferredBlockPersistOp = func(blockID flow.Identifier, tx *Tx) error { return nil }
```

Comment:
```suggestion
const noOpPersist DeferredBlockPersistOp = func(blockID flow.Identifier, tx *Tx) error { return nil }
```

Reply:
linter doesn't like that:
```
storage/badger/transaction/deferred_block_persist.go:18:19: invalid constant type DeferredBlockPersistOp (typecheck)
```
Not entirely sure that [this description](https://go.dev/tour/basics/15) is exhaustive:
> Constants can be character, string, boolean, or numeric values.